src/convert/impls.rs

```diff
@@ -624,3 +625,123 @@ where
     }
     result.into_boxed_slice()
 }
+
+// WebAssembly ToWebAssemblyValue implementations for primitive types
+// These implement the exact WebAssembly specification algorithms
+
+impl JsValueCast for i32 {
+    fn unchecked_from_js_value(js_value: JsValue) -> Self {
+        // WebAssembly ToInt32(v) algorithm
+        let number = js_value.unchecked_into_f64();
+        if !number.is_finite() || number == 0.0 || number == -0.0 {
```

Comment:
`number == 0.0 || number == -0.0` is redundant (just use number == 0.0). Comparison with zero is sign-agnostic according to IEEE754